File: backend/test/integration/setup/seed.go
package setup

// TestUsersData provides easy access to details of users seeded by seed.sql
var TestUsersData = map[string]struct {
	Email             string
	PlainTextPassword string
	UserID            string
	TenantID          string
	Name              string
	Role              string
	Status            string
}{
	"alpha_admin": {
		Email:             "alpha_admin@example.com",
		PlainTextPassword: "password123",
		UserID:            "b0000000-0000-0000-0000-000000000001",
		TenantID:          "a0000000-0000-0000-0000-000000000001",
		Name:              "Alpha Admin",
		Role:              "admin",
		Status:            "active",
	},
	"alpha_editor": {
		Email:             "alpha_editor@example.com",
		PlainTextPassword: "password123",
		UserID:            "b0000000-0000-0000-0000-000000000002",
		TenantID:          "a0000000-0000-0000-0000-000000000001",
		Name:              "Alpha Editor",
		Role:              "editor",
		Status:            "active",
	},
	"pending_user_valid_token": {
		Email:             "pending_user_valid_token@example.com",
		PlainTextPassword: "password", // Initial password before activation
		UserID:            "b0000000-0000-0000-0000-000000000003",
		TenantID:          "a0000000-0000-0000-0000-000000000001",
		Name:              "Pending User Valid Token",
		Role:              "editor",
		Status:            "pending_verification",
	},
	"suspended_user": {
		Email:             "suspended_user@example.com",
		PlainTextPassword: "password123",
		UserID:            "b0000000-0000-0000-0000-000000000004",
		TenantID:          "a0000000-0000-0000-0000-000000000001",
		Name:              "Suspended User",
		Role:              "editor",
		Status:            "suspended",
	},
	"beta_admin": {
		Email:             "beta_admin@example.com",
		PlainTextPassword: "password123",
		UserID:            "b0000000-0000-0000-0000-000000000005",
		TenantID:          "a0000000-0000-0000-0000-000000000002",
		Name:              "Beta Admin",
		Role:              "admin",
		Status:            "active",
	},
}

File: frontend/test/e2e/setup/seed.ts
export const TestUsersData = {
	alpha_admin: {
		email: "alpha_admin@example.com",
		plainTextPassword: "password123", // As per comment in seed.sql
		userID: "b0000000-0000-0000-0000-000000000001",
		tenantID: "a0000000-0000-0000-0000-000000000001",
		name: "Alpha Admin",
		role: "admin",
		status: "active"
	},
	alpha_user: {
		email: "alpha_user@example.com",
		plainTextPassword: "password123",
		userID: "b0000000-0000-0000-0000-000000000002",
		tenantID: "a0000000-0000-0000-0000-000000000001",
		name: "Alpha User",
		role: "editor",
		status: "active"
	},
	pending_editor_valid_token: {
		email: "pending_user_valid_token@example.com",
		plainTextPassword: "password123", // seed.sql comment says 'password123', Go file says 'password'. Assuming seed.sql is source of truth for hash.
		userID: "b0000000-0000-0000-0000-000000000003",
		tenantID: "a0000000-0000-0000-0000-000000000001",
		name: "Pending User Valid Token",
		role: "editor",
		status: "pending_verification" // As per seed.sql
	},
	suspended_user: {
		email: "suspended_user@example.com",
		plainTextPassword: "password123",
		userID: "b0000000-0000-0000-0000-000000000004",
		tenantID: "a0000000-0000-0000-0000-000000000001",
		name: "Suspended User",
		role: "editor",
		status: "suspended"
	},
	beta_admin: {
		email: "beta_admin@example.com",
		plainTextPassword: "password123",
		userID: "b0000000-0000-0000-0000-000000000005",
		tenantID: "a0000000-0000-0000-0000-000000000002",
		name: "Beta Admin",
		role: "admin",
		status: "active"
	}
} as const;

export type TestUserKey = keyof typeof TestUsersData;
export type TestUser = (typeof TestUsersData)[TestUserKey];

// Example Tenant Data (if useful for tests)
export const TestTenantsData = {
	tenant_alpha: {
		id: "a0000000-0000-0000-0000-000000000001",
		name: "Tenant Alpha",
		plan: "basic"
	},
	tenant_beta: {
		id: "a0000000-0000-0000-0000-000000000002",
		name: "Tenant Beta",
		plan: "basic"
	}
} as const;

export type TestTenantKey = keyof typeof TestTenantsData;
export type TestTenant = (typeof TestTenantsData)[TestTenantKey];
